### PR TITLE
Update checksum verification

### PR DESCRIPTION
**Pull request by gfcas**
@koecher So here is the promised update of the checksum verification. As the macos CI is fine, this feature should be supported also for macos. However, at some point (maybe later) we should discuss a strategy for return/exit commands?

In this commit the checksum algorithm can be either md5, sha1 or sha256. In the verify_archive function the length of the checksum is queried in an if-block. The archive is verified with the first valid checksum. An example is given in the p4est package, where additional checksums are defined. In order to check first the most secure one, the sha256sum is the first in the list. For backward compatibility: It is not necessary to define additional checksums.

### CODE REVIEW

File: candi.sh [hunk 1]
```diff
@@ -202,7 +202,12 @@ quit_if_fail() {
 #  return 1: No checksum provided          (archive found, but unable to verify)
 #  return 2: ARCHIVE_FILE not found        (archive NOT found)
 #  return 3: CHECKSUM mismatch             (archive file corrupted)
-#  return 4: Neither md5 nor md5sum found  (archive found, but unable to verify)
+#  return 4: Not able to compute checksum  (archive found, but unable to verify)
+#  return 5: Checksum algorithm not found  (archive found, but unable to verify)
+# This function tries to verify the downloaded archive by determing and
+# comapring checksums. For a specific package several checksums might be
```
Comment (koecher) [inline candi.sh:208]: spelling: comparing

File: candi.sh [hunk 2]
```diff
@@ -224,32 +229,62 @@ verify_archive() {
 
     # Skip verifying archive, if CHECKSUM=ignore
     if [ "${CHECKSUM}" = "ignore" ]; then
-        cecho ${WARN} "Skipped checksum check for ${ARCHIVE_FILE}"
+        cecho ${WARN} "Ignore checksum check for ${ARCHIVE_FILE}"
```
Comment (koecher) [inline candi.sh:232]: well, "ignore" comes from my side and "skip" is/was the naming from dorsal. Maybe we should use CHECKSUM="skip" and the original cecho

**Review by koecher — APPROVED** [review]
Only minor comments. Very nice.

File: candi.sh (hunk 2 above)
Comment (gfcas) [reply candi.sh:232]: The remaining 'ignore' found by 'git grep ignore' belong to BUILDCHAIN=ignore.

**Comment by gfcas** [comment]
Thanks for your review @koecher!

**Comment by koecher** [comment]
I'm not sure, but do all packages currently have at least one checksum? (Hopefully we don't have ones with CHECKSUM=ignore)

This PR is ready to be merged I think

**Comment by gfcas** [comment]
No, no CHECKSUM=ignore were found by git grep ignore.

**Comment by koecher** [comment]
@gfcas perfect, thanks for checking that! The ubuntu testers are also happy...

**Pull request merged**